htlcswitch/mailbox.go

```diff
@@ -397,22 +430,56 @@ func (m *memoryMailBox) mailCourier(cType courierType) {
 			}
 
 		case pktCourier:
+			var (
+				pktOutbox chan *htlcPacket
+				addOutbox chan *htlcPacket
+			)
+
+			// Prioritize delivery of Settle/Fail packets over Adds.
+			// This ensures that we actively clear the commitment of
+			// existing HTLCs before trying to add new ones. This
+			// can help to improve forwarding performance since the
+			// time to sign a commitment is linear in the number of
+			// HTLCs manifested on the commitments.
+			//
+			// NOTE: Both types are eventually delivered over the
+			// same channel, but we can control which is delivered
+			// by exclusively making one nil and the other non-nil.
+			// We know from our loop condition that at least one
+			// nextPkt and nextAdd are non-nil.
+			if nextPkt != nil {
+				pktOutbox = m.pktOutbox
+			} else {
+				addOutbox = m.pktOutbox
+			}
+
 			select {
-			case m.pktOutbox <- nextPkt:
+			case pktOutbox <- nextPkt:
```

Comment:
Hmm, is it possible that this actually can panic given one of them will be nil? Or does this also leverage the feature of Go where select case statements involving `nil` channels will always block? (so not be taken if other paths are available)

Reply:
Alternatively, we could make the two channels above buffered, and send to them, only to read from either of them here (with the read only succeeding for the packet queue that was actually populated).

Reply:
> Or does this also leverage the feature of Go where select case statements involving nil channels will always block? (so not be taken if other paths are available)

Correct the `nil` channels will just block, it won't panic bc the channel isn't closed